File: Other/TurboPower Flash Filer/examples/cbuilder/ExOrderU.cpp
//---------------------------------------------------------------------------

#include <vcl\vcl.h>
#pragma hdrstop

#include "ExOrderU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "ffdb"
#pragma link "ffdbbase"
#pragma link "ffllbase"
#pragma link "ffclreng"
#pragma link "ffllcomm"
#pragma link "ffllcomp"
#pragma link "fflleng"
#pragma link "fflllgcy"
#pragma link "ffsrintm"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::LinesTableCalcFields(TDataSet *DataSet)
{
  LinesTableTotal->AsFloat =
    LinesTablePrice->AsFloat * LinesTableCount->AsInteger;
}
//---------------------------------------------------------------------------

void __fastcall TForm1::Open1Click(TObject *Sender)
{
  ProductTable->Active = true;
  CustomerTable->Active = true;
  OrdersTable->Active = true;
  LinesTable->Active = true;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Close1Click(TObject *Sender)
{
  LinesTable->Active = false;
  OrdersTable->Active = false;
  CustomerTable->Active = false;
  ProductTable->Active = false;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Exit1Click(TObject *Sender)
{
  Close1Click(0);
  Close();
}
//---------------------------------------------------------------------------

File: Torry's Delphi Pages/Other/tpflashfiler/examples/delphi/exorderu.pas
(* ***** BEGIN LICENSE BLOCK *****
 * Version: MPL 1.1
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 1.1 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * The Original Code is TurboPower FlashFiler
 *
 * The Initial Developer of the Original Code is
 * TurboPower Software
 *
 * Portions created by the Initial Developer are Copyright (C) 1996-2002
 * the Initial Developer. All Rights Reserved.
 *
 * Contributor(s):
 *
 * ***** END LICENSE BLOCK ***** *)
 
 unit ExOrderu;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Db, StdCtrls, Grids, DBGrids, FFDB, FFDBBase, Menus, ffllbase, ffllcomm,
  fflllgcy, ffllcomp, fflleng, ffsrintm, ffclreng;

type
  TForm1 = class(TForm)
    ffSess: TffSession;
    CustomerData: TDataSource;
    LinesData: TDataSource;
    OrdersData: TDataSource;
    CustomerTable: TffTable;
    OrdersTable: TffTable;
    LinesTable: TffTable;
    ProductTable: TffTable;
    LinesTableLineID: TIntegerField;
    LinesTableOrderID: TIntegerField;
    LinesTableProductID: TIntegerField;
    LinesTableCount: TIntegerField;
    LinesTableTotal: TCurrencyField;
    MainMenu1: TMainMenu;
    File1: TMenuItem;
    Open1: TMenuItem;
    Close1: TMenuItem;
    N1: TMenuItem;
    Exit1: TMenuItem;
    GroupBox1: TGroupBox;
    DBGrid1: TDBGrid;
    GroupBox2: TGroupBox;
    DBGrid4: TDBGrid;
    DBGrid3: TDBGrid;
    LinesTableDescription: TStringField;
    LinesTablePrice: TCurrencyField;
    ffClient: TffClient;
    ffRSE: TFFRemoteServerEngine;
    ltMain: TffLegacyTransport;
    procedure LinesTableCalcFields(DataSet: TDataSet);
    procedure Open1Click(Sender: TObject);
    procedure Close1Click(Sender: TObject);
    procedure Exit1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.LinesTableCalcFields(DataSet: TDataSet);
begin
  LinesTableTotal.AsFloat :=
    LinesTablePrice.AsFloat * LinesTableCount.AsInteger;
end;

procedure TForm1.Open1Click(Sender: TObject);
begin
  ProductTable.Active := True;
  CustomerTable.Active := True;
  OrdersTable.Active := True;
  LinesTable.Active := True;
end;

procedure TForm1.Close1Click(Sender: TObject);
begin
  LinesTable.Active := False;
  OrdersTable.Active := False;
  CustomerTable.Active := False;
  ProductTable.Active := False;
end;

procedure TForm1.Exit1Click(Sender: TObject);
begin
  Close1Click(nil);
  Close;
end;

end.
